sample-app/src/androidTest/kotlin/com/liferay/apio/blog/postings/activity/BlogPostingListTest.kt

```diff
@@ -74,7 +74,9 @@ class BlogPostingListTest {
 		val url = HttpUrl.parse("http://screens.liferay.org.es/o/api/p/groups/57459/blogs")
 
 		url?.let {
-			ApioConsumer(BasicAuthenticator(credentials)).fetch(url, mapOf(), listOf(), {
+			ApioConsumer.setAuthenticator(BasicAuthenticator(credentials))
```

Comment:
this API is strange... I would prefer it inside fetch, what do you think @victorg1991 ?

Reply:
Agree, or maybe in the constructor too, with the ability to customize it in each fetch call. In the same fashion that the apio consumer js

Reply:
The other problem we had was trying to perform operations before a fetch, so we hadn't the right credentials set. 

I'll change it back to a class, adding the credentials param in the constructor and in functions.

Reply:
Ok, to that I'll send another PR taking care of the states for each instance. For now, it keeps the same behavior as before.